mpop_kiosk/UserProfile.qml

```diff
@@ -210,6 +209,8 @@ Item {
                                 });
                             }
                         });
+                        // Set the userId property, so that we can use it from anywhere
+                        userId = user_id;
```

Comment:
Nice catch! I think it should be move within the callback for when we call _populateUserAnswers. (on line 201 in your PR)